src/ReflectionFileNamespace.php

```diff
@@ -83,6 +83,9 @@ class ReflectionFileNamespace
      */
     public function __construct($fileName, $namespaceName, Namespace_ $namespaceNode = null)
     {
+        if (!is_string($fileName)) {
```

Comment:
It seems, that you've started to change more code. I'd recommend doing this in another PRs.